### PR TITLE
`make sagemath_categories-check`, `make pypi-wheels-check`

### PR DESCRIPTION
**Pull request by mkoeppe**
We add `make` targets for testing wheel-building script packages that have already been installed. For example:
```
$ make SAGE_WHEELS=yes sagemath_categories  # builds and stores (but does not install) a wheel
$ make sagemath_categories-check            # tests the wheel using tox
```
The target `make pypi-wheels-check` calls all of them. We restructure the Build & Test CI so that all tests are run after build has been completed. Later, as a follow up of this and #36446, we can store the result of Build as a container image and then parallelize the various tests on top of it.

This is implemented on top of an improvement of how we record information on our wheels.
```
$ cat venv/var/lib/sage/scripts/sagemath_categories/spkg-requirements.txt  
sagemath_categories @ file:///Users/mkoeppe/s/sage/sage-rebasing/worktree-algebraic-2018-spring/local/var/lib/sage/venv-python3.11/var/lib/sage/wheels/sagemath_categories-10.2b6-cp311-cp311-macosx_13_0_x86_64.whl
```
Previously, this information was only known during the execution of the `spkg-install` script.

In a follow-up, we can address #30956 by delaying the wheel installation to spkg-postinst.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -165,6 +164,25 @@ jobs:
           MAKE: make -j2 --output-sync=recurse
           SAGE_NUM_THREADS: 2
 
+      # Testing
+
+      - name: Test changed files (sage -t --new)
+        run: |
+          # We run tests with "sage -t --new"; this only tests the uncommitted changes.
```
Comment (kwankyu) [inline .github/workflows/build.yml:172]: `if: always() && steps.worktree.outcome == 'success'`?

Comment (mkoeppe) [reply .github/workflows/build.yml:172]: thanks, i've made a similar change in 8224cb0

**Comment by kwankyu** [comment]
Thus previously 
```
make SAGE_CHECK=yes sagemath_categories                    
```
worked, but will not work after this PR. Instead we will use 
```
make sagemath_categories-check                                           
```
Am I right?

If so, then do we need to update some relevant part of the developer guide?

**Comment by mkoeppe** [comment]
> Thus previously
> 
> ```
> make SAGE_CHECK=yes sagemath_categories                    
> ```
> 
> worked, but will not work after this PR

No, it still works the same way. The makefile invokes the `spkg-check` script after running the `spkg-install`.

File: build/bin/sage-spkg [hunk 2]
```diff
@@ -567,24 +567,6 @@ for script in $WRAPPED_SCRIPTS; do
...
-    elif [ -f setup.py ]; then
-        echo 'python setup.py install' >> spkg-install
-    else
-        echo >&2 "Error: There is no spkg-install script, no setup.py, and no configure"
-        echo >&2 "script, so I do not know how to install $PKG_SRC."
-        exit 1
-    fi
-    chmod +x spkg-install
-fi
-
```
Comment (kwankyu) [inline build/bin/sage-spkg:587]: Is this never used, and so removed?

**Comment by kwankyu** [comment]
OK. It looks good as far as I can tell.

It works well as described.

**Review by kwankyu — APPROVED** [review]

File: build/bin/sage-spkg (hunk 2 above)
Comment (mkoeppe) [reply build/bin/sage-spkg:587]: Yes, that's right. It was a leftover from the abandoned support for "old-style packages"

**Comment by mkoeppe** [comment]
Thank you!

**Pull request merged**